concourse/steps/release.py

```diff
@@ -576,21 +576,25 @@ def name(self):
 
     def validate(self):
         version.parse_to_semver(self.release_version)
+
         # either cds _XOR_ ctf must exist
         have_ctf = os.path.exists(self.ctf_path)
         have_cd = os.path.exists(self.component_descriptor_v2_path)
         if not have_ctf ^ have_cd:
             ci.util.fail('exactly one of component-descriptor, or ctf-archive must exist')
         elif have_cd:
-            component_descriptor_v2 = cm.ComponentDescriptor.from_dict(
+            self.components = [cm.ComponentDescriptor.from_dict(
                 ci.util.parse_yaml_file(self.component_descriptor_v2_path),
-            )
-            # resolve all referenced components (thus ensure all dependencies are available)
-            tuple(cnudie.retrieve.components(component=component_descriptor_v2))
-            # TODO: more validation (e.g. check for uniqueness of names)
+            )]
         elif have_ctf:
-            # nothing to do, already uploaded in component_descriptor step.
-            pass
+            self.components = list(component_descriptors_from_ctf_archive(
+                self.ctf_path,
+            ))
+            if not component_descriptors:
+                ci.util.fail(f'No component descriptor found in CTF archive at {self.ctf_path}')
```

Comment:
nit: `{self.ctf_path=}` will also print the variable-name (helpful for debugging)